source/android/adaptivecards/src/androidTest/java/io/adaptivecards/objectmodel/TextBlockPropertiesTest.java

```diff
@@ -307,6 +309,19 @@ public void WrapTest() throws Exception
         }
     }
 
+    @Test
+    public void TestLineBreaks () throws Exception
+    {
+        final String textWithNewLines = "This is some string\nAnd this is another line\r";
+        // This looks counter intuitive but without the replacement of '\n\r' for "<br/>" the
+        // output will only contain a blank space where '\n' is expected
+        final String expectedHtml = "This is some string\nAnd this is another line ";
+
+        String html = RendererUtil.handleSpecialText(textWithNewLines).toString();
+
+        Assert.assertEquals(expectedHtml, html);
```

Comment:
Nit: Maybe add a quick descriptive text message here to debug failures? "'\r' failed to be parsed as a newline" for eg?